cpp/include/cudf/detail/aggregation/aggregation.cuh

```diff
@@ -314,6 +314,32 @@ struct update_target_element<dictionary32, aggregation::SUM, target_has_nulls, s
   }
 };
 
+template <typename Source, bool target_has_nulls, bool source_has_nulls>
+struct update_target_element<Source,
+                             aggregation::SUM_OF_SQUARES,
+                             target_has_nulls,
+                             source_has_nulls,
+                             std::enable_if_t<is_numeric<Source>()>> {
+  __device__ void operator()(mutable_column_device_view target,
+                             size_type target_index,
+                             column_device_view source,
+                             size_type source_index) const noexcept
+  {
+// This code will segfault in nvcc/ptxas 10.2 only
+// https://nvbugswb.nvidia.com/NvBugs5/SWBug.aspx?bugid=3186317
+// Enabling only for 2 types does not segfault. Using for unit tests.
+#if (__CUDACC_VER_MAJOR__ == 10) and (__CUDACC_VER_MINOR__ == 2)
+    if (not std::is_floating_point<Source>::value) return;
```

Comment:
So will this just yield silently incorrect results on 10.2? I think we should detect this and throw.

Reply:
Updated. modified to throw.